assets/prometheus-k8s/rules.yaml

```diff
@@ -611,6 +611,23 @@ spec:
       record: count:up1
     - expr: count without(instance, pod, node) (up == 0)
       record: count:up0
+  - name: openshift-general.rules
+    rules:
+    - alert: TargetDown
+      annotations:
+        description: '{{ printf "%.4g" $value }}% of the {{ $labels.job }}/{{ $labels.service
+          }} targets in {{ $labels.namespace }} namespace have been unreachable for
+          more than 60 minutes. This may be a symptom of network connectivity issues,
+          down nodes, or failures within these components. Assess the health of the
+          infrastructure and nodes running these targets and then contact support.'
+        summary: Some targets were not reachable from the monitoring server for an
+          extended period of time.
+      expr: |
+        100 * (count(up == 0 unless on (node) max by (node) (kube_node_spec_unschedulable == 1)) BY (job, namespace, service) /
+          count(up unless on (node) max by (node) (kube_node_spec_unschedulable == 1)) BY (job, namespace, service)) > 10
+      for: 60m
```

Comment:
Why 60 minutes? The point of a smaller window is to make sure your workloads are monitored, otherwise warning the admin something is wrong and they cannot rely on existing alerts they setup for those workloads.

Reply:
Because this is a general catch all for every sort of other disruption.  At a minimum, some disruption can occur for a very long time sequentially (such as a node restart).  Looking at fleet data 90% of these alerts are still firing 60m later, so reducing chattiness probably isn’t going to have impact.  “All targets down” would be critical and fast - a few down isn’t a big deal (if you are alreayd running multiple)